### ISSUE TITLE
The important of num classes argument 

### ISSUE THREAD

**Issue by thangngoc89** [Qsingle/LearnablePromptSAM#3]
May I ask the importance of the num classes argument? Without seeing example training data, it’s hard to know why the default is 12 classes. TIA

**Comment by Qsingle** [comment]
We do not use the mask decoder, so we use one segmentation head to fine-tune the model. So we must tell the task head how many classes it needs to output. Because in one of our tasks, the number of types is 12. If you want to use the code in the current version, you can set it based on your task.

**Comment by thangngoc89** [comment]
@Qsingle Thank you for your reply. In my case, each image has about 10-14 structures of the same class (the exact number is unknown unless I run some object detection algorithm over first), should the number of classes be 10-14 or one?

**Comment by Qsingle** [comment]
> @Qsingle Thank you for your reply. In my case, each image has about 10-14 structures of the same class (the exact number is unknown unless I run some object detection algorithm over first), should the number of classes be 10-14 or one?

If you want to classify each structure as an individual class, you can set it to 10-14. Or you can set it to 2 (see the background as one class).

**Comment by thangngoc89** [comment]
> > @Qsingle Thank you for your reply. In my case, each image has about 10-14 structures of the same class (the exact number is unknown unless I run some object detection algorithm over first), should the number of classes be 10-14 or one?
> 
> If you want to classify each structure as an individual class, you can set it to 10-14. Or you can set it to 2 (see the background as one class).

Thank you very much for your answers and for releasing the code
